lib/3scale/backend/async_redis/client.rb

```diff
@@ -31,32 +31,49 @@ def url_present?(url)
           def connect_tcp(opts)
             uri = URI(opts[:url])
 
-            credentials = [ uri.user || opts[:username], uri.password || opts[:password]]
-            db = uri.path[1..-1]
-
-            protocol = Protocol::ExtendedRESP2.new(db: db, credentials: credentials)
+            database = uri.path[1..-1]
+            credentials = [ uri.user || opts[:username], uri.password || opts[:password]].compact
+            credentials = nil unless credentials.any?
+            ssl = opts[:ssl]
+            ssl_params = opts[:ssl_params]
+            limit = opts[:max_connections]
 
             if opts.key? :sentinels
-              SentinelsClientACLTLS.new(uri, protocol, opts)
+              ssl_context = EndpointHelpers.create_ssl_context(ssl:, ssl_params:)
+
+              master_options = { database:, credentials:, ssl_context:}.compact
+              master_name = uri.host
+              role = opts[:role] || :master
+              endpoints = opts[:sentinels].map do |sentinel|
+                host = sentinel[:host]
+                port = sentinel[:port]
+                sentinel_credentials = [opts[:sentinel_username], opts[:sentinel_password]].compact
+                sentinel_credentials = nil unless sentinel_credentials.any?
```

Comment:
Just a nit
```suggestion
                sentinel_credentials = [opts[:sentinel_username], opts[:sentinel_password]].compact.presence
```

Reply:
`presence` is a Rails method, we don't have it in apisonator.